server/lib/chat/chat-models.ts

```diff
@@ -247,80 +237,96 @@ export async function getMessagesForChannel(
 
 export interface LeaveChannelResult {
   /**
-   * The ID of a new owner of the channel, or null if the channel ownership has been left unchanged.
+   * The ID of a user that was selected as a new owner of the channel, or `undefined` if the channel
+   * ownership has been left unchanged.
    */
-  newOwnerId: SbUserId | null
+  newOwnerId?: SbUserId
 }
 
 export async function removeUserFromChannel(
   userId: SbUserId,
   channelName: string,
 ): Promise<LeaveChannelResult> {
   return transact(async function (client) {
-    const channelUserResult = await client.query<DbUserChannelEntry>(sql`
+    await client.query(sql`
       DELETE FROM channel_users
-      WHERE user_id = ${userId} AND channel_name = ${channelName}
-      RETURNING *`)
-    if (channelUserResult.rowCount < 1) {
-      throw new Error('No rows returned')
-    }
+      WHERE user_id = ${userId} AND channel_name = ${channelName};
+    `)
 
-    let result = await client.query(sql`
+    const deleteChannelResult = await client.query(sql`
       DELETE FROM channels
       WHERE name = ${channelName} AND
         NOT EXISTS (SELECT 1 FROM channel_users WHERE channel_name = ${channelName})
-      RETURNING name`)
-    if (result.rowCount > 0) {
+      RETURNING name;
+    `)
+    if (deleteChannelResult.rowCount > 0) {
       // Channel was deleted; meaning there is no one left in it so there is no one to transfer the
       // ownership to
-      return { newOwnerId: null }
+      return { newOwnerId: undefined } as LeaveChannelResult
```

Comment:
Seems like these casts would be unnecessary if you just left the key off? (e.g. `return {}`)